fix: date parsing functions: case-insensitive name parsing

tomersela opened the pull request:
fixes #7961

### Description 
This PR changes `PARSE_TIMESTAMP`, `PARSE_DATE` to parse case-insensitive strings.
As written in the [original issue](https://github.com/confluentinc/ksql/issues/7961) `PARSE_TIMESTAMP('29-JUN-21','dd-MMM-yy')` yields a null value, but will now result with the same value as of `PARSE_TIMESTAMP('29-Jun-21','dd-MMM-yy')`.

Affected functions - `PARSE_TIMESTAMP`, `PARSE_DATE` (and the deprecated functions - `STRINGTOTIMESTAMP`, `STRINGTODATE`)

### Testing done 
Wrote unit tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #7961")

tomersela commented:
[clabot:check]

jzaralim reviewed (commented):
Thank you for the fix @tomersela, this looks good! Just two things:

1. Do you mind also adding the fix to `ParseTime`? AM/PM is also affected. You test with this test case: `udf.parseTime("12:01:05 aM", "hh:mm:ss a")`
2. We have another set of tests called `QueryTranslationTests`. The relevant test files are [parse-date.json](https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/parse-date.json), [parse-time.json](https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/parse-time.json) and [stringtimestamp.json](https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/stringtimestamp.json). After adding to these files, you will need to un-ignore and run PlannedTestGeneratorTest. This generates test cases to ensure backwards compatibility. Instructions for running these tests are in the [README](https://github.com/confluentinc/ksql/tree/master/ksqldb-functional-tests) under ksqldb-functional-tests.

tomersela commented:
Thanks for the review @jzaralim 
1. I added a fix to ParseTime + test. 
2. Updated the JSON files you mentioned with new cases and verified it runs successfully with -
```
./mvnw test -pl ksqldb-functional-tests -Dtest=QueryTranslationTest -Dksql.test.files=parse-date.json,parse-time.json,stringtimestamp.json
```

jzaralim reviewed (approved):
Thank you, this LGTM! Community builds are temporarily disabled, but I'll ping you once that opens up again.

agavra commented:
@jzaralim there's a workaround where we can kick off a PR with the same commit, if that is green then this PR will turn green as well. I can do that

agavra commented:
ah turns out the workaround doesn't work because the author of the commit is different, thereby changing the hash :( I guess we'll have to wait

tomersela commented:
Thanks for checking this out @agavra 

@jzaralim I'd be happy to contribute more to the project once the community builds are back. Thanks.

jzaralim commented:
@tomersela The build from #8040 passed, so I'll just go ahead and merge this. Thanks again for the fix!
